app/src/main/java/com/islaharper/dawnofandroid/domain/model/CustomColourScheme.kt

```diff
@@ -0,0 +1,44 @@
+package com.islaharper.dawnofandroid.domain.model
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.staticCompositionLocalOf
+import androidx.compose.ui.graphics.Color
+import com.islaharper.dawnofandroid.ui.theme.DarkStarBackgroundColour
+import com.islaharper.dawnofandroid.ui.theme.DarkStarColour
+import com.islaharper.dawnofandroid.ui.theme.LightStarBackgroundColour
+import com.islaharper.dawnofandroid.ui.theme.LightStarColour
+
+/* https://developer.android.com/jetpack/compose/designsystems/custom
+    Add additional custom colours as needed
+    Usage: DawnOfAndroidTheme.colourScheme.myColour
+*/
+
+@Immutable
+data class CustomColourScheme(
```

Comment:
Maybe I'm wrong here, but I was thinking more like a solution like the following:

```kotlin
object RatingBarDefaults {

    @Composable
    fun colors(): RatingBarColors {
        return if (isSystemInDarkTheme()) {
            RatingBarColors(
                starColor = Color(0x),
                starBackgroundColor = Color(0x)
            )
        } else {
            RatingBarColors(
                starColor = Color(0x),
                starBackgroundColor = Color(0x)
            )
        }
    }
}

data class RatingBarColors(
    val starColor: Color,
    val starBackgroundColor: Color
)

@Composable
fun RatingBar(
    rating: Double,
    modifier: Modifier = Modifier,
    spaceBetween: Dp = PADDING_SMALL,
    colors: RatingBarColors = RatingBarDefaults.colors()
) {
```

Reply:
My initial thinking was that additional colours could be added if needed and accessed in a similar way to MaterialTheme.colorScheme, but I do like that your solution provides better context in the composable for the colours used.